astropy/table/tests/test_column.py

```diff
@@ -361,15 +362,28 @@ def test_insert_string_masked_values(self):
         c2 = c.insert(1, np.ma.MaskedArray(["ccc", "dd"], mask=[True, False]))
         assert np.all(c2 == ["a", "ccc", "dd", "b"])
         assert np.all(c2.mask == [False, True, False, False])
-        assert c2.dtype == "U3"
+        if NUMPY_LT_2_0:
+            expected_c2_dtype = "U3"
+        else:
+            expected_c2_dtype = np.dtype("<U2")
+        assert c2.dtype == expected_c2_dtype
 
     def test_insert_string_type_error(self, Column):
         c = Column([1, 2])
         with pytest.raises(ValueError, match="invalid literal for int"):
             c.insert(0, "string")
 
         c = Column(["a", "b"])
-        with pytest.raises(TypeError, match="string operation on non-string array"):
+        if NUMPY_LT_2_0:
+            ctx = pytest.raises(TypeError, match="string operation on non-string array")
+        else:
+            from numpy._core._exceptions import _UFuncNoLoopError
+
+            ctx = pytest.raises(
+                _UFuncNoLoopError,
+                match="ufunc 'str_len' did not contain a loop with signature matching types",
```

Comment:
Since `UFuncNoLoopError` is a subclass of `TypeError`, only the match needs to be changed, i.e.,
```
with pytest.raises(TypeError, match="string operation on non-string array" if NUMPY_LT_2_0 else "ufunc 'str_len' did not contain a loop"):
    ...
```

Reply:
I've done this as part of what I'll push in a sec; squashing my changes with your third commit.